### PR TITLE
Fix compatibility issue with DoctrineModule 5.2.0

### PR DESCRIPTION
**Pull request by driehle**
Since DoctrineMongoODMModule 4.0.x is officially not maintained anymore, I am releasing a fix to mark this module as incompatible to DoctrineModule >= 5.2.0. 

For the 4.1.x series I am hoping to get this running again with a new release of DoctrineModule.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -53,7 +53,7 @@
         "php": "^7.4 || ~8.0.0 || ~8.1.0",
         "ext-mongodb": "*",
         "container-interop/container-interop": "^1.2.0",
-        "doctrine/doctrine-module": "^5.0.0",
+        "doctrine/doctrine-module": "^5.0.0 <5.2.0",
```
Comment (Ocramius) [inline composer.json:56]: Shouldn't there be a comma here? :thinking:

Comment (driehle) [reply composer.json:56]: According to https://getcomposer.org/doc/articles/versions.md#version-range:

> You can define multiple ranges. Ranges separated by a space ( ) or comma (,) will be treated as a logical AND. A double pipe (||) will be treated as a logical OR. AND has higher precedence than OR.

So it shouldn't matter :-)

**Comment by driehle** [comment]
There we go, CI back working 💪 

The change causing the issue was in DoctrineModule 5.2.0 in `DriverFactory.php`, see https://github.com/doctrine/DoctrineModule/compare/5.1.1...5.2.0

**Review by Ocramius — APPROVED** [review]
👍

**Review by malarzm — APPROVED** [review]
I don't really understand the changes but am a simple man: I see @Ocramius' apprive I'll give one too :)

**Pull request merged**